code/game/objects/structures/displaycase.dm

```diff
@@ -396,64 +396,135 @@
 	src.icon = I
 	return
 
+/obj/structure/displaycase/forsale/ui_interact(mob/user, ui_key, datum/tgui/ui, force_open, datum/tgui/master_ui, datum/ui_state/state)
+	. = ..()
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "Vendatray", name, 300, 300, master_ui, state)
+		ui.open()
+
+/obj/structure/displaycase/forsale/ui_data(mob/user)
+	var/list/data = list()
+	var/register = FALSE
+	if(payments_acc)
+		register = TRUE
+		data["owner_name"] = payments_acc.account_holder
+	if(showpiece)
+		data["product_name"] = capitalize(showpiece.name)
+		var/base64 = icon2base64(icon(showpiece.icon, showpiece.icon_state))

```

Comment:
This is expensive. Very bad idea to add this in ui_data of processing UI. Needs a cache and ideally making this static/ui not auto processing.